lib/input.js

```diff
@@ -2,17 +2,22 @@
 var config = require('./config.js'),
     algo = require('./algorithms.js');
     async = require('async');
-
-var updateSequence = function(userId, itemId, callback){
-  algo.updateSimilarityFor(userId, function(){
+    
+var updateSequence = function(userId, itemId, update, callback){
+  //make the update parameter optional while still allowing the callback to be last (for readability)
+  if (typeof update == 'function') {
+    callback = update;
+    update = true;
```

Comment:
Why not make use of ES6's [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)?